docs: update docs for idle connection timeout (#8021)
confluentinc/ksql: docs/reference/server-configuration.md

After hitting the limit, the host will fail pull query requests until it determines that it's no longer
at the limit.

## `ksql.idle.connection.timeout.seconds`

Sets the timeout for idle connections. A connection is idle if there is no data in either direction
on that connection for the duration of the timeout. This configuration can be helpful if you are 
issuing push queries that only receive data infrequently from the server, as otherwise those

> connections will be severed when the timeout (default 10min) is hit.

Reviewer: `10min` if it's a string literal, or "10 min" (no quotes) if it's just text

> Decreasing this timeout allows you to close connections more aggressively and save server resources

Reviewer: ```suggestion
Decreasing this timeout enables closing connections more aggressively to save server resources.
```

> while increasing this timeout makes the server more tolerant of low-data volume use cases.

Reviewer: ```suggestion
Increasing this timeout makes the server more tolerant of low-data volume use cases.
```
